### PR TITLE
Resolve some leaks in user code

### PR DESCRIPTION
**Pull request by benharsh**
This PR updates some tests such that they no longer leak (or leak less). The highlights include:
- not break-ing from a loop (leaking iter classes)
- adding dsiDestroy* methods to a non-standard domain map
- more use of ``borrowed`` over ``unmanaged``

Testing:
- [x] local + futures
- [x] gasnet + futures
- [x] memleaks (down to ~25K leaked)

### CODE REVIEW

File: test/associative/bharshbarg/arrays/diffDomZip.chpl [hunk 1]
```diff
@@ -29,7 +29,6 @@ while !different {
   for (aa, bb) in zip(A,B) do
     if aa != bb {
       different = true;
-      break;
```
Comment (cassella) [inline test/associative/bharshbarg/arrays/diffDomZip.chpl]: Is removing the break just a workaround, or is there something wrong with breaking from a loop like this?

Comment (benharsh) [reply test/associative/bharshbarg/arrays/diffDomZip.chpl]: Yeah, it's just a workaround.

**Comment by benharsh** [comment]
@mppf (if you're the right person) : would you mind reviewing the UserMapAssoc changes?
@daviditen : would you mind reviewing the changes under ``test/studies/hpcc/RA/diten/`` ?

File: test/distributions/bradc/assoc/UserMapAssoc.chpl [hunk 2]
```diff
@@ -508,6 +508,12 @@ class UserMapAssocDom: BaseAssociativeDom {
   proc dsiReprivatize(other) {
     locDoms = other.locDoms;
   }
+
+  proc dsiDestroyDom() {
+    for localeIdx in dist.targetLocDom do
```
Comment (mppf) [inline test/distributions/bradc/assoc/UserMapAssoc.chpl]: Why is this a 'for' while destroyArr uses a coforall?

**Comment by mppf** [comment]
I had a minor question about UserMapAssoc changes but the rest look good to me.

**Comment by daviditen** [comment]
The RA changes look fine to me.

I don't think you should remove the `break` in `diffDomZip.chpl`.  We should fix the underlying problem instead of changing the behavior or tests to work around it.

File: test/distributions/bradc/assoc/UserMapAssoc.chpl (hunk 2 above)
Comment (benharsh) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: Good point, not sure why I did that...

Comment (mppf) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: The pull of the leak-free side is strong and hard to resist.

Comment (mppf) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: Is it that way in something you based it off of, Block etc?

Comment (benharsh) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: Ah, I think I based it on the loop in UserMapAssocDom.setup. Anyways, I'll update it to use a coforall and re-run tests on that directory.

Comment (mppf) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: I asked because I wondered if whatever you based it on should use coforall.

Comment (mppf) [reply test/distributions/bradc/assoc/UserMapAssoc.chpl]: Maybe worth a comment if it's not obvious.

**Pull request merged**